python/social_bot/envs/gazebo_base.py

```diff
@@ -49,6 +51,54 @@ def render(self, mode='human'):
         raise NotImplementedError(
             "rendering mode 'rgb_array' is not implemented.")
 
+    def _get_internal_states(self, agent, agent_joints):
+        joint_pos = []
+        joint_vel = []
+        for joint_id in range(len(agent_joints)):
+            joint_name = agent_joints[joint_id]
+            joint_state = agent.get_joint_state(joint_name)
+            joint_pos.append(joint_state.get_positions())
+            joint_vel.append(joint_state.get_velocities())
+        joint_pos = np.array(joint_pos).flatten()
+        joint_vel = np.array(joint_vel).flatten()
+        # pos of continous joint could be huge, wrap the range to [-pi, pi)
+        joint_pos = (joint_pos + np.pi) % (2 * np.pi) - np.pi
+        internal_states = np.concatenate((joint_pos, joint_vel), axis=0)
+        return internal_states
+
+    def _construct_dict_space(self, obs_sample, vocab_size):
+        """
+        Args:
+            obs_sample (dict) : a sample observation
+            vocab_size (int): the vocab size for the sequence space
+        Returns:
+            Return a gym.spaces.Dict with keys 'image', 'states', 'sequence'
+            Possible situation:
+                image with internal states
+                image with language sequence
+                image with both internal states and language sequence
+                pure low-dimensional states with language sequence
+        """
+        ob_space_dict = dict()
+        if 'image' in obs_sample.keys():
+            ob_space_dict['image'] = gym.spaces.Box(
+                low=0,
+                high=255,
+                shape=obs_sample['image'].shape,
+                dtype=np.uint8)
+        if 'states' in obs_sample.keys():
+            ob_space_dict['states'] = gym.spaces.Box(
+                low=-np.inf,
+                high=np.inf,
+                shape=obs_sample['states'].shape,
+                dtype=np.float32)
+        if 'sequence' in obs_sample.keys():
```

Comment:
'sentence' might be a better name for this field as it reflects the actual meaning more accurately.

Reply:
Yes. Sentence is better. We can name the un-ecoded one sentence_raw. sentence_string or something. Any suggestion?

Reply:
Updated, using sentence_raw and sentence_seq as the name of variables for the 2 different domain .